src/dbnode/environment/config.go

```diff
@@ -24,30 +24,32 @@ import (
 	"errors"
 	"fmt"
 
+	"github.com/m3db/m3/src/dbnode/kvconfig"
+
 	clusterclient "github.com/m3db/m3/src/cluster/client"
 	etcdclient "github.com/m3db/m3/src/cluster/client/etcd"
 	"github.com/m3db/m3/src/cluster/kv"
 	m3clusterkvmem "github.com/m3db/m3/src/cluster/kv/mem"
 	"github.com/m3db/m3/src/cluster/services"
 	"github.com/m3db/m3/src/cluster/shard"
-	"github.com/m3db/m3/src/dbnode/kvconfig"
 	"github.com/m3db/m3/src/dbnode/sharding"
 	"github.com/m3db/m3/src/dbnode/storage/namespace"
 	"github.com/m3db/m3/src/dbnode/topology"
 	"github.com/m3db/m3x/instrument"
 )
 
 var (
-	errInvalidConfig = errors.New("must supply either service or static config")
+	errInvalidConfig    = errors.New("must supply either service or static config")
+	errInvalidSyncCount = errors.New("must supply exactly one synchronous cluster")
 )
 
 // Configuration is a configuration that can be used to create namespaces, a topology, and kv store
 type Configuration struct {
-	// Service is used when a topology initializer is not supplied.
-	Service *etcdclient.Configuration `yaml:"service"`
+	// Services is used when a topology initializer is not supplied.
+	Services DynamicConfiguration `yaml:"services"`
 
-	// StaticConfiguration is used for running M3DB with a static config
-	Static *StaticConfiguration `yaml:"static"`
+	// StaticConfiguration is used for running M3DB with static configs
+	Static *StaticConfiguration `yaml:"statics"`
```

Comment:
Hm this will be likely a breaking change. 

It might be easier to reuse some of the work in the query/m3coordinator config to initiate multi cluster clients (although right now each cluster needs a concept of resolution and retention, but we could abstract some of that away and keep it in that package but move the core multi-cluster stuff elsewhere):
https://github.com/m3db/m3/blob/master/src/query/storage/m3/config.go#L169-L314

Reply:
I've [overridden `UnmarshalYAML`](https://github.com/m3db/m3/pull/1859/files/1515a5f3353934320d591e9c3014d888d6d1c80b#diff-66a5adec93dca45d214b25e038ce1d3aR130) so the `service` key is still valid, I'll need to do the same thing for statics. 

Will take a look through the coordinator and see how it might fit in!